test/e2e/bootstrap/corefile_template.go

```diff
@@ -68,6 +68,7 @@ data:
 	})
 
 	E2EAfterAll(func() {
+		DebugCPLogs(k8sCluster)
```

Comment:
That's not the way. It should be in 

It should more something like:
```
	AfterEachFailure(func() {
		DebugUniversal(cluster, "default")
	})
```

because the outcome would be a bit different in that case.

Reply:
The `JustAfterEach` ensure it's really the first teardown stuff that happens.

Reply:
Replaced with `AfterEachFailure`